### PR TITLE
NonPrintableCharacterFixer - Add option to replace with escape sequences

### PR DESCRIPTION
**Pull request by julienfalque**
Using this option makes the fixer non-risky by replacing characters with `\u{...}` escape sequences instead of removing them.

I also replaced `pack('CCC', x, y, z)` with `pack('H*', 'xyz')` because I find the later easier to read, is it ok?

### CODE REVIEW

File: src/Fixer/Basic/NonPrintableCharacterFixer.php [hunk 1]
```diff
@@ -88,11 +98,72 @@ public function isCandidate(Tokens $tokens)
...
      */
+    protected function createConfigurationDefinition()
+    {
+        return new FixerConfigurationResolver([
+            (new FixerOptionBuilder('use_escape_sequences_in_strings', 'Whether characters should be replaced with escape sequences in strings.'))
+                ->setAllowedTypes(['bool'])
+                ->setDefault(false)
+                ->setNormalizer(function (Options $options, $value) {
+                    if (PHP_VERSION_ID < 70000 && $value) {
+                        throw new InvalidOptionsException('Escape sequences require PHP 7.0+.');
```
Comment (keradus) [inline src/Fixer/Basic/NonPrintableCharacterFixer.php]: please add it to php7.0 migration:risky set

File: src/Fixer/Basic/NonPrintableCharacterFixer.php [hunk 2]
```diff
@@ -88,11 +98,72 @@ public function isCandidate(Tokens $tokens)
     /**
      * {@inheritdoc}
      */
+    protected function createConfigurationDefinition()
+    {
+        return new FixerConfigurationResolver([
+            (new FixerOptionBuilder('use_escape_sequences_in_strings', 'Whether characters should be replaced with escape sequences in strings.'))
+                ->setAllowedTypes(['bool'])
+                ->setDefault(false)
```
Comment (keradus) [inline src/Fixer/Basic/NonPrintableCharacterFixer.php]: please add a TODO to change it to true on 3.0

Comment (julienfalque) [inline src/Fixer/Basic/NonPrintableCharacterFixer.php]: Done.

File: src/Fixer/Basic/NonPrintableCharacterFixer.php (hunk 1 above)
Comment (julienfalque) [inline src/Fixer/Basic/NonPrintableCharacterFixer.php]: Why risky? The new option is supposed to make the fixer non-risky.

Comment (keradus) [inline src/Fixer/Basic/NonPrintableCharacterFixer.php]: cool, right !

File: README.rst [hunk 3]
```diff
@@ -743,6 +743,11 @@ Choose from the list of available rules:
 
   *Risky rule: risky when strings contain intended invisible characters.*
 
+  Configuration options:
```
Comment (keradus) [inline README.rst]: old comment is outdated for some reason, please adjust the ruleset

File: src/Fixer/Basic/NonPrintableCharacterFixer.php (hunk 1 above)
Comment (keradus) [inline src/Fixer/Basic/NonPrintableCharacterFixer.php]: indeed, add it to php7.0 migration, non-risky one

Comment (julienfalque) [inline src/Fixer/Basic/NonPrintableCharacterFixer.php]: Done.

**Comment by keradus** [comment]
Thank you @julienfalque.

**Pull request closed without merging**